src/js/models/layerstructure.js

```diff
@@ -1540,5 +1551,24 @@ define(function (require, exports, module) {
         return this._setTextStyleProperties("paragraphStyle", layerIDs, properties);
     };
 
+    /**
+     * Given a set of layers representing the current state, overlay the visibility status
+     * of any matching layers
+     *
+     * TODO is there a way to do this with mergeWith?
+     *
+     * @param {LayerStructure} layerTree
+     * @return {LayerStructure}
+     */
+    LayerStructure.prototype.overlayWithLayers = function (layerTree) {
+        var nextLayers = layerTree.layers.map(function (layer, key) {
```

Comment:
Can't you use `mergeDeep` here if `layerTree only has `<id: {id, visible}>` mapping?